packages/aws-cdk-lib/aws-cloudtrail/lib/cloudtrail.ts

```diff
@@ -262,6 +270,22 @@ export class Trail extends Resource {
       },
     }));
 
+    if (props.isOrganizationTrail) {
+      this.s3bucket.addToResourcePolicy(new iam.PolicyStatement({
+        resources: [this.s3bucket.arnForObjects(
```

Comment:
In the doc, it mentions `"arn:aws:s3:::myBucketName/[optionalPrefix]/AWSLogs/myAccountID/*"`. I haven't looked deep into it, but is the optional prefix part of the bucket and it would be handled by `this.s3bucket.arnForObjects`?

Reply:
It would be this value for the resources: `"arn:aws:s3:::myOrganizationBucket/AWSLogs/o-organizationID/*"` where `arn:aws:s3:::myOrganizationBucket/` is handled by `this.s3bucket.arnForObjects`. But for the case you're talking about it would not handle it, [the optional prefix part is passed in from props](https://github.com/aws/aws-cdk/blob/d30881882f0742362d171340d2f94be8b7707acc/packages/aws-cdk-lib/aws-cloudtrail/lib/cloudtrail.ts#L254-L257):
```
    this.s3bucket.addToResourcePolicy(new iam.PolicyStatement({
      resources: [this.s3bucket.arnForObjects(
        `${props.s3KeyPrefix ? `${props.s3KeyPrefix}/` : ''}AWSLogs/${Stack.of(this).account}/*`,
      )],
```